pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/documentation/CommentContentRule.java

```diff
@@ -4,152 +4,74 @@
 
 package net.sourceforge.pmd.lang.java.rule.documentation;
 
-import static net.sourceforge.pmd.properties.PropertyFactory.booleanProperty;
-import static net.sourceforge.pmd.properties.PropertyFactory.stringListProperty;
+import static net.sourceforge.pmd.properties.PropertyFactory.regexProperty;
 
 import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Locale;
-import java.util.Set;
+import java.util.regex.Pattern;
 
-import org.apache.commons.lang3.StringUtils;
-
-import net.sourceforge.pmd.RuleContext;
+import net.sourceforge.pmd.lang.document.Chars;
 import net.sourceforge.pmd.lang.java.ast.ASTCompilationUnit;
-import net.sourceforge.pmd.lang.java.ast.Comment;
+import net.sourceforge.pmd.lang.java.ast.JavaComment;
 import net.sourceforge.pmd.lang.java.rule.AbstractJavaRulechainRule;
 import net.sourceforge.pmd.properties.PropertyDescriptor;
-import net.sourceforge.pmd.properties.PropertySource;
 
 /**
  * A rule that checks for illegal words in the comment text.
  *
- * TODO implement regex option
- *
  * @author Brian Remedios
  */
 public class CommentContentRule extends AbstractJavaRulechainRule {
 
-    private boolean caseSensitive;
-    private List<String> originalBadWords;
-    private List<String> currentBadWords;
-
-    // ignored when property above == True
-    public static final PropertyDescriptor<Boolean> CASE_SENSITIVE_DESCRIPTOR = booleanProperty("caseSensitive").defaultValue(false).desc("Case sensitive").build();
-
-    public static final PropertyDescriptor<List<String>> DISSALLOWED_TERMS_DESCRIPTOR =
-            stringListProperty("disallowedTerms")
-                    .desc("Illegal terms or phrases")
-                    .defaultValues("idiot", "jerk").build(); // TODO make blank property? or add more defaults?
-
-    private static final Set<PropertyDescriptor<?>> NON_REGEX_PROPERTIES;
-
-    static {
-        NON_REGEX_PROPERTIES = new HashSet<>(1);
-        NON_REGEX_PROPERTIES.add(CASE_SENSITIVE_DESCRIPTOR);
-    }
+    private static final PropertyDescriptor<Pattern> DISSALLOWED_TERMS_DESCRIPTOR =
+        regexProperty("forbiddenRegex")
```

Comment:
We should add a note in 7_0_0_release_notes.md under "Changed Rules": For "CommentContent" the properties "caseSensitive" and "disallowedTerms" are replaced by the regex property "fobiddenRegex".